configure

```diff
@@ -179,59 +235,70 @@ done
 
 # Sanity checks
 
-case "$prefix" in
+case $prefix in
   /*) ;;
    *) err "The -prefix directory must be absolute.";;
 esac
-case "$bindir" in
+case $bindir in
   /*) ;;
   "") ;;
-  '$(PREFIX)/'*) ;;
+  '$(PREFIX)/'*) bindir=$prefix/${bindir#'$(PREFIX)/'};;
    *) err 'The -bindir directory must be absolute or relative to $(PREFIX).';;
 esac
-case "$libdir" in
+case $libdir in
   /*) ;;
   "") ;;
-  '$(PREFIX)/'*) ;;
+  '$(PREFIX)/'*) libdir=$prefix/${libdir#'$(PREFIX)/'};;
    *) err 'The -libdir directory must be absolute or relative to $(PREFIX).';;
 esac
-case "$mandir" in
+case $mandir in
   /*) ;;
   "") ;;
-  '$(PREFIX)/'*) ;;
+  '$(PREFIX)/'*) mandir=$prefix/${mandir#'$(PREFIX)/'};;
    *) err 'The -mandir directory must be absolute or relative to $(PREFIX).';;
 esac
 
 # Generate the files
 
 cd config/auto-aux
 rm -f s.h m.h Makefile
-touch s.h m.h Makefile
+touch s.h m.h
+
+makefile=4
+
+exec 4>Makefile
 
 # Write options to Makefile
+printf %s\\n "# generated by ./configure$configure_options" |
+    sed 's/$/\\/
+$ s|\\$|\
+# ifeq ($(flavor EMPTY),undefined)\
+# $(error Include $$(TOPDIR)/global.mk instead of $$(TOPDIR)/config/Makefile)\
+# endif\
+override EMPTY :=|
+' >&4
```

Comment:
Several things wrong with these lines:
1. using `sed` to append to a file in a roundabout way.
2. using `\` at the end of a comment line.
3. using `EMPTY` to tell whether `global.mk` was included.